stdlib/sql/to.go

```diff
@@ -0,0 +1,361 @@
+				q := fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (%s)", t.spec.Spec.Table, strings.Join(newSQLTableCols, ","))
+				_, err = t.tx.Exec(q)
+				if err != nil {
+					if rbErr := t.tx.Rollback(); err != nil {
+						return fmt.Errorf("\n%s\n%s", err, rbErr)
+					}
+					return err
+				}
+			}
+
+			for i := 0; i < l; i++ {
+				valueStrings = append(valueStrings, valuePlaceHolders)
+				for j, col := range er.Cols() {
+					switch col.Type {
+					case flux.TFloat:
+						if er.Floats(j).IsNull(i) {
+							valueArgs = append(valueArgs, nil)
+							break
+						}
+						valueArgs = append(valueArgs, er.Floats(j).Value(i))
+					case flux.TInt:
+						if er.Ints(j).IsNull(i) {
+							valueArgs = append(valueArgs, nil)
+							break
+						}
+						valueArgs = append(valueArgs, er.Ints(j).Value(i))
+					case flux.TUInt:
+						if er.UInts(j).IsNull(i) {
+							valueArgs = append(valueArgs, nil)
+							break
+						}
+						valueArgs = append(valueArgs, er.UInts(j).Value(i))
+					case flux.TString:
+						if er.Strings(j).IsNull(i) {
+							valueArgs = append(valueArgs, nil)
+							break
+						}
+						valueArgs = append(valueArgs, er.Strings(j).ValueString(i))
+					case flux.TTime:
+						if er.Times(j).IsNull(i) {
+							valueArgs = append(valueArgs, nil)
+							break
+						}
+						valueArgs = append(valueArgs, values.Time(er.Times(j).Value(i)).Time())
+					case flux.TBool:
+						if er.Bools(j).IsNull(i) {
+							valueArgs = append(valueArgs, nil)
+							break
+						}
+						valueArgs = append(valueArgs, er.Bools(j).Value(i))
+					default:
+						return fmt.Errorf("invalid type for column %s", col.Label)
+					}
+				}
+
+				if err := execute.AppendRecord(i, er, builder); err != nil {
+					return err
+				}
+
+				if i != 0 && i%BatchSize == 0 {
+					if err := ExecuteQueries(t.tx, t.spec.Spec, colNames, &valueStrings, &valueArgs, TestingWriter); err != nil {
+						return err
+					}
+				}
+			}
+			if len(valueArgs) > 0 && len(valueStrings) > 0 {
+				if err := ExecuteQueries(t.tx, t.spec.Spec, colNames, &valueStrings, &valueArgs, TestingWriter); err != nil {
+					return err
+				}
+			}
+			return nil
+		})
+		return err
+	})
+
+	if err := wg.Wait(); err != nil {
+		return err
+	}
+
+	return err
+}
+
+func (t *ToSQLTransformation) UpdateWatermark(id execute.DatasetID, pt execute.Time) error {
+	return t.d.UpdateWatermark(pt)
+}
+
+func (t *ToSQLTransformation) UpdateProcessingTime(id execute.DatasetID, pt execute.Time) error {
+	return t.d.UpdateProcessingTime(pt)
+}
+
+func (t *ToSQLTransformation) Finish(id execute.DatasetID, err error) {
+	if t.spec.Spec.DriverName != "sqlmock" {
+		var txErr error
+		if err == nil {
+			txErr = t.tx.Commit()
+		} else {
+			txErr = t.tx.Rollback()
+		}
+		if txErr != nil {
+			panic(txErr)
+		}
+	}
+	t.d.Finish(err)
+}
+
+func ExecuteQueries(tx *sql.Tx, s *ToSQLOpSpec, colNames []string, valueStrings *[]string, valueArgs *[]interface{}, writer *Writer) error {
```

Comment:
Is it possible to remove `writer` from this signature? It seems like it's only included for testing purposes?